whisper/utils.py

```diff
@@ -159,9 +160,18 @@ def iterate_subtitles():
                         if last != start:
                             yield last, start, subtitle_text
 
+                        if subtitle_format == "bold":
+                            reg_pattern = r"\1<b>\2</b>"
+                        
+                        elif subtitle_format == "italic":
+                            reg_pattern = r"\1<i>\2</i>"
+
+                        else:
+                            reg_pattern = r"\1<u>\2</u>"
+
```

Comment:
Note that if the parameter were simply `b`, `i`, or `u` (i.e. the highlight tag), then you can do a direct substitution without having 3 if cases for 3 patterns.

That said, probably the goal here is to offer a selection of standard interchange formats, and not necessarily to provide stylistic rendering preferences. Do we know what the most popular encoding for Karaoke subtitles is? If there is a most popular one, my vote would be for that one, since you can then import that file into other tools to do post editing for stylistic preferences.

Reply:
Thanks for getting back to me, I was trying to submit this question to the Whisper author, but wasn't sure where I should submit it. So I'll submit the question here. Since I don't really know how to program, I can't really understand what you are talking about. Thanks again for your reply!

Reply:
> Note that if the parameter were simply `b`, `i`, or `u` (i.e. the highlight tag), then you can do a direct substitution without having 3 if cases for 3 patterns.
> 
> That said, probably the goal here is to offer a selection of standard interchange formats, and not necessarily to provide stylistic rendering preferences. Do we know what the most popular encoding for Karaoke subtitles is? If there is a most popular one, my vote would be for that one, since you can then import that file into other tools to do post editing for stylistic preferences.

Thank you for your review and insightful comments regarding the style format parameter in the pull request.

Indeed, Advanced SubStation Alpha (ASS) format is often acknowledged as one of the most comprehensive and versatile formats for karaoke subtitles. ASS, being an extension of SubStation Alpha (SSA), offers an extensive range of styling options, encompassing advanced text formatting, precise positioning, a wide array of color choices, animations, and various visual effects. Given its rich feature set, ASS is widely favored for creating intricate and visually captivating karaoke subtitles. 

Open source tools like [Aegisub](https://aegisub.org/) can be used then to post edit video with stylistic preferences.

Reply:
I probably caused some confusion with the way I phrased that question, but I was referring to the 3 styling options you are proposing for SRT karaoke files. I would vote for providing only whichever one is the most widely used out of the 3, with the intention of outputting a standard interchange format that can then be imported into other tools to customise the styling further. ASS indeed has a lot of rendering flexibility (too much to be useful as an interchange format for Karaoke subtitles IMO), but there are tools to convert SRT to ASS and then customise the style.

Reply:
> I probably caused some confusion with the way I phrased that question, but I was referring to the 3 styling options you are proposing for SRT karaoke files. I would vote for providing only whichever one is the most widely used out of the 3, with the intention of outputting a standard interchange format that can then be imported into other tools to customise the styling further. ASS indeed has a lot of rendering flexibility (too much to be useful as an interchange format for Karaoke subtitles IMO), but there are tools to convert SRT to ASS and then customise the style.

It makes sense to streamline things and go for the most commonly used styling option out of the three. That way, we're aiming for a standard interchange format that can easily be imported into other tools for further customization.

I totally agree that ASS offers a ton of rendering flexibility, perhaps a bit too much for the purpose of an interchange format for Karaoke subtitles. Converting SRT to ASS and then customizing the style seems like a viable approach, considering the needs of the karaoke community. Karaoke usually involves a mix of word highlighting, often in green or yellow, and some kind of bold style. So, having an option that covers these aspects should cater to the majority of karaoke scenarios with the Whisper model.

Reply:
May I ask, do you know where I can give feedback on bugs to the Whisper Ai creative team? Thank you!

Reply:
You're on the pull request page now, and that's the place to come to if you want to fix a bug and contribute that back to Whisper. For everything else, you can go to the "Discussions" page. There are many existing discussions already so you could check if that bug has already been discussed, otherwise start a new discussion.

Reply:
> Note that if the parameter were simply `b`, `i`, or `u` (i.e. the highlight tag), then you can do a direct substitution without having 3 if cases for 3 patterns.
> 
> That said, probably the goal here is to offer a selection of standard interchange formats, and not necessarily to provide stylistic rendering preferences. Do we know what the most popular encoding for Karaoke subtitles is? If there is a most popular one, my vote would be for that one, since you can then import that file into other tools to do post editing for stylistic preferences.

Hi, I would like to try this method of yours, but don't know anything about programming. Can you tell me the details of how to do it? I am really running Whisper in Colab. thank you very much!